lib/api/helpers/namespaces.rb

```diff
@@ -8,6 +8,10 @@ def can_manage_namespace?(namespace, user)
         NamespacePolicy.new(user, namespace).update?
       end
 
+      def can_destroy_namespace?(namespace, user)
+        NamespacePolicy.new(user, namespace).destroy? && !User.exists?(namespace: namespace)
```

Comment:
It worked partially. Just need to check for global namespaces to deny that too.

Reply:
That is true. We are not checking this in neither the policy nor the service. Where should we put this check :thinking: ? I'd say that you could argue for both in this case:

1. If you put it on the policy, it means that no one is *authorized* to delete a global namespace. It means that this will never happen (e.g. it's not like the personal namespace case because in that case we may eventually allow that).
2. If you put it on the service, it means that what you are trying to do is semantically wrong, regardless of your credentials.

On this I would favor option 1, but I'm not sure. @vitoravelino what do you think?

Reply:
@mssola option 1 sounds better.